### PR TITLE
mailing: add subscribe endpoint for jobs weekly

### PR DESCRIPTION
**Pull request by drjova**
* INSPIR-2632

Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: backend/inspirehep/mailing/providers/mailchimp.py [hunk 1]
```diff
@@ -78,5 +78,38 @@ def send_mailchimp_campaign(campaign_id, html_content, test_emails=None):
...
+
+
+def mailchimp_subscribe_user_to_list(list_id, email, first_name, last_name):
+    """Subscribe user to a list.
+
+    Args:
+        list_id (str): the list id .
+        email (str): user email.
+        email (str): user first name.
+        email (str): user last name.
```
Comment (ammirate) [inline backend/inspirehep/mailing/providers/mailchimp.py]: copy/paste

File: backend/inspirehep/mailing/providers/mailchimp.py [hunk 2]
```diff
@@ -78,5 +78,38 @@ def send_mailchimp_campaign(campaign_id, html_content, test_emails=None):
...
+    """
+    client = mailchimp_get_client()
+
+    LOGGER.info("Adding user to list.", list_id=list_id, email=email)
```
Comment (ammirate) [inline backend/inspirehep/mailing/providers/mailchimp.py]: `... to mailing list.` ? :)

File: backend/inspirehep/mailing/views.py [hunk 3]
```diff
@@ -5,9 +5,31 @@
...
+            data["email"], data["first_name"], data["last_name"]
+        )
+        LOGGER.info("User successfuly subscribed.", user=data)
+        return jsonify({"message": "Succesfully subscribed."}), 200
+    except ValueError:
+        LOGGER.info("Invalid email address.", user=data)
+        return jsonify({"message": "Not a valid email address."}), 400
+    except Exception:
+        LOGGER.exception("Cannot subscribe user to list.", user=data)
+        return jsonify({"message": "Unexpected error."}), 500
```
Comment (ammirate) [inline backend/inspirehep/mailing/views.py]: Is it possible to get a `KeyError`? In that case the error message can be more helpful, like `Missing field email`

Comment (drjova) [reply backend/inspirehep/mailing/views.py]: @ammirate updated with Marshmallow validation

**Pull request merged**